src/input.rs

```diff
@@ -437,6 +532,27 @@ mod tests {
         }
     }
 
+    test_clickstate! {
+        name: single_click,
+        initial_state: ClickState::None,
+        input: Event::MouseInput(ElementState::Pressed, MouseButton::Left),
+        end_state: ClickState::Click
+    }
+
+    test_clickstate! {
+        name: double_click,
+        initial_state: ClickState::Click,
+        input: Event::MouseInput(ElementState::Pressed, MouseButton::Left),
+        end_state: ClickState::DblClick
```

Comment:
Handler methods are in the same block as the `MouseState` state bump. Should I move those state transitions into the handler functions to ensure that the handler functions are being called?